supertokens/supertokens-auth-react
# Fixes for how component is calculated

## Summary of change

`getMatchingComponentForRouteAndRecipeIdFromPreBuiltUIList` did not correctly fetch the components for non auth recipe. So that has been fixed.

## Test Plan
Without this change, the way to reproduce the issue was to:
- Enable dynamic login method
- first factor is set to email password in tenant
- second factor is set to otp-email
- Load the tenant, login into first factor, and then see that the second factor screen keeps redirecting to auth

NOTE: This PR has not implemented any test case

## Checklist for important updates

-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [x] Had run `npm run build-pretty`
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If added a new recipe interface, then make sure that the implementation of it uses NON arrow functions only (like `someFunc: function () {..}`).
-   [ ] If I added a new recipe, I also added the recipe entry point into the `size-limit` section of `package.json` with the size limit set to the current size rounded up.
-   [ ] If I added a new recipe, I also added the recipe entry point into the `rollup.config.mjs`
-   [ ] If I added a new login method, I modified the list in `lib/ts/types.ts`
-   [ ] If I added a factor id, I modified the list in `lib/ts/recipe/multifactorauth/types.ts`
